cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/ToolCommand.java

```diff
@@ -305,45 +313,58 @@ private void handleUseCommand() {
                 return;
             }
-            CommandUtil.printError(errStream, "tool '" + toolId + ":" + version + "' is not found. " +
-                    "Run 'bal tool pull " + toolId + ":" + version
-                    + "' to fetch and set as the active version.", null, false);
+            CommandUtil.printError(errStream, "tool '" + toolId + ":" + version + "' is not found. Run " +
+                            "'bal tool pull " + toolId + ":" + version + "' to fetch and set as the active version.",
+                    null, false);
             CommandUtil.exitError(this.exitWhenFinish);
             return;
         }
         this.org = tool.get().org();
         this.name = tool.get().name();
-        Optional<BalToolsManifest.Tool> currentActiveTool = balToolsManifest.getActiveTool(toolId);
+        Optional<BalToolsManifest.Tool> currentActiveTool = blendedBalToolsManifest.getActiveTool(toolId);
         if (currentActiveTool.isPresent() && currentActiveTool.get().version().equals(tool.get().version()) &&
                 Objects.equals(currentActiveTool.get().repository(), tool.get().repository())) {
             outStream.println("tool '" + toolId + ":" + version + "' is the current active version.");
             return;
         }
 
-        boolean isDistsCompatible = checkToolDistCompatibility();
-        if (!isDistsCompatible) {
+        boolean isCompatibleWithPlatform = isCompatibleWithPlatform(
+                org, name, version, tool.get().repository());
+        if (!isCompatibleWithPlatform) {
             CommandUtil.exitError(this.exitWhenFinish);
             return;
         }
 
-        balToolsManifest.setActiveToolVersion(toolId, version, repositoryName);
+        balToolsManifest.resetCurrentActiveVersion(toolId);
+        if (!DISTRIBUTION_REPOSITORY_NAME.equals(tool.get().repository())) {
+            balToolsManifest.setActiveToolVersion(toolId, version, tool.get().repository());
+        }
         balToolsToml.modify(balToolsManifest);
         outStream.println("tool '" + toolId + ":" + version + "' successfully set as the active version.");
     }
 
     private void handleListCommand() {
-        if (argList.size() > 1) {
+        if (argList.size() > 2) {
             CommandUtil.printError(
                     this.errStream, "too many arguments.", TOOL_LIST_USAGE_TEXT, false);
             CommandUtil.exitError(this.exitWhenFinish);
             return;
         }
-        List<BalToolsManifest.Tool> tools = listBalToolsTomlFile();
+        boolean all = false;
+        if (argList.size() == 2) {
+            if (!ALL.equals(argList.get(1))) {
+                CommandUtil.printError(
+                        this.errStream, "invalid argument.", TOOL_LIST_USAGE_TEXT, false);
+                CommandUtil.exitError(this.exitWhenFinish);
+                return;
+            }
+        }
+        List<BalToolsManifest.Tool> tools = listBalToolsTomlFile(all);
```

Comment:
Isn't the value of `all` always false here?

Reply:
Addressed in [b2b3f93](https://github.com/ballerina-platform/ballerina-lang/commit/b2b3f93fba80d28c5b244baf3b6754b662eb872f)